microservice-app/src/main/java/com/sdevelopment/skt/microservice/dao/impl/ProductRepositoryImpl.java

```diff
@@ -0,0 +1,84 @@
+package com.sdevelopment.skt.microservice.dao.impl;
+
+import com.sdevelopment.skt.common.domain.Product;
+import com.sdevelopment.skt.microservice.dao.ProductRepository;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.persistence.StoredProcedureQuery;
+import java.util.List;
+
+public class ProductRepositoryImpl implements ProductRepository {
+
+    @PersistenceContext
+    private EntityManager em;
+
+    @Override
+    public void saveProduct(Product product) {
+        StoredProcedureQuery saveProductProcedure =
+                em.createNamedStoredProcedureQuery("saveProduct");
+        saveProductProcedure.execute();
+    }
+
+    @Override
+    public List<Product> getAllProducts() {
+        StoredProcedureQuery findAllProcedure =
+                em.createNamedStoredProcedureQuery("getAllProducts");
+        return findAllProcedure.getResultList();
+    }
+
+    @Override
+    public <S extends Product> S save(S s) {
+        return null;
+    }
+
+    @Override
+    public <S extends Product> Iterable<S> save(Iterable<S> iterable) {
+        return null;
+    }
+
+    @Override
+    public Product findOne(Long aLong) {
```

Comment:
it is not correct to implement all this functionality, as this is already provided by `CrudRepository` interface.
This needs to be removed by fixing `ProductRepository` interface. 

Please check @emoteNS comment in `ProductRepositoryImpl.java`

Reply:
Already fixed after refactoring.